### ISSUE TITLE
Error Message format does not comply with rfc9457 (Problem Details)

### ISSUE THREAD

**Issue by jason-fox** [ScorpioBroker/ScorpioBroker#509]
Looking at the 1.7.1 NGSI-LD specification:

>   **5.5.3 Error response payload body**
>
> When reporting errors back to clients, NGSI-LD implementations shall generate a JSON object in accordance with
> IETF RFC 7807 [10], section 3.1, including, at least the following terms:
>  -  `type`: Error type as per clause 5.5.2.
>  -  `title`: Error title which shall be a short string summarizing the error.
>  -  `detail`: A detailed message that should convey enough information about the error.
>
> Even though IETF RFC 7807 [10] defines a specific MIME type for error payloads, NGSI-LD implementations shall
> use the standard JSON MIME type ("application/json") when reporting errors, so that old clients or existing tools are
> not broken. 

[RFC 7809](https://www.rfc-editor.org/rfc/rfc7807.txt) has been obsoleted by [RFC 9457,](https://www.rfc-editor.org/rfc/rfc9457.txt) but the following structure remains true:

>    A problem details object can have the following members:
> 
>    -  `type` (string) - A URI reference [RFC3986] that identifies the problem type.  This specification encourages that, when dereferenced, it provide human-readable documentation for the problem type (e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be   "about:blank".
> 
>    -  `title` (string) - A short, human-readable summary of the problem type.  It SHOULD NOT change from occurrence to occurrence of the problem, except for purposes of localization (e.g., using  proactive content negotiation; see [RFC7231], Section 3.4).
>    -  `status` (number) - The HTTP status code ([RFC7231], Section 6)  generated by the origin server for this occurrence of the problem.
>    -  `detail` (string) - A human-readable explanation specific to this occurrence of the problem.
>    -  `instance` (string) - A URI reference that identifies the specific occurrence of the problem.  It may or may not yield further information if dereferenced.

A current Scorpio Error message looks like this:

```json
{
  "type":"https://uri.etsi.org/ngsi-ld/errors/ResourceNotFound",
  "title":"Resource not found.",
   "detail": {
        "message":"urn:ngsi-ld:Building:farm001 was not found"
   },
  "errorCode":404
}
```

According to the IETF RFCs, it should look like this:

```json
{
  "type":"https://uri.etsi.org/ngsi-ld/errors/ResourceNotFound",
  "title":"Resource not found.",
  "detail":"urn:ngsi-ld:Building:farm001 was not found",
  "status": 404,
  "instance": "urn:ngsi-ld:Building:farm001"
}
```

- `detail` is defined as a string, not an Object with a `message` attribute.
- `errorCode` should be called `status`
-  `instance` is an **Optional** field according the IETF, but in the example 404, the URI of the entity would be the preferred term.

**Comment by martin-p-bauer** [comment]
Indeed the value of `detail` should be just a string. Instead of `errorCode`, `status` should be used, but according to RFC 9457, it should be a number (same as currently in `errorCode`), i.e. without the quote. `instance` would be a relevant addition. We need to check, in which cases it makes sense.

**Comment by jason-fox** [comment]
> without the quote

Yes. corrrected.

**Comment by jason-fox** [comment]
Fixed 4.1.15
